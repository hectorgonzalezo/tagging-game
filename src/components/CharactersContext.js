import { createContext } from 'react';

// This will contain all the character to be found in the current MainImage;
const CharactersContext = createContext([]);

export default CharactersContext;
