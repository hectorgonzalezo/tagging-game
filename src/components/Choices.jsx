import React, { useState, useContext } from 'react';
import styled from 'styled-components';
import uniqid from 'uniqid';
import { func } from 'prop-types';
import CharactersContext from './CharactersContext';
import CharacterOption from './CharacterOption';

const Options = styled.div`
  position: absolute;
  background-color: var(--midnight-green-eagle-green-3);
  left: 50%;
  transform: translate(-50%, 0);
  bottom: 110%;
  padding: 10px;
  border: 3px solid white;
  border-radius: 1rem;
  color: white;
  display: grid;
  grid-template-columns: 1fr 1fr;
  grid-template-rows: 1fr;
  gap: 7px;

  animation: grow 1s;

  & > button:first-child{
    background-color: red;
    grid-column: 1 / -1;
    justify-self: start;
    padding: 5px;

    :hover{
      background-color: var(--palatinate-purple);
    }
  }

  @keyframes grow {
  from {
    transform: translate(-50%, 0) scale(0.1);
    opacity: 0;
  }
  to {
    transform: translate(-50%, 0) scale(1);
    opacity: 1;
  }
`;

// Make an option for each character
function Choices({ closeTarget }) {
  const characters = useContext(CharactersContext);
  return (
    <Options>
      <CharacterOption name="x" onClick={closeTarget} />
      {characters.map((character) => (
        <CharacterOption key={character + uniqid()} name={character} onClick={()=>{}}/>
      ))}
    </Options>
  );
}

Choices.propTypes = {
  closeTarget: func.isRequired,
};

export default Choices;
