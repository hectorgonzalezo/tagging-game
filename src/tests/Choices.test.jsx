import React from 'react';
import { render, screen, container } from '@testing-library/react';
import '@testing-library/jest-dom';
import userEvent from '@testing-library/user-event';
import Choices from '../components/Choices';
import CharactersContext from '../components/CharactersContext';

describe('Choices between characters', () => {
  test('Choices should be empty if context stays as initialized', async () => {
    const mockClose = jest.fn();

    render(<Choices closeTarget={mockClose} location={{ x: 10, y: 20 }} />);

    const closeButton = screen.getByRole('button', { name: 'x' });
    // there should be no option rendered, only the close button
    expect(closeButton).toBeInTheDocument();
    expect(screen.queryByRole('button')).toEqual(closeButton);
    expect(container).toMatchSnapshot();
  });

  test('When updating context, there should be a CharacterOption', async () => {
    const mockClose = jest.fn();
    render((
      <div>
        <CharactersContext.Provider value={['juan']}>
          <Choices closeTarget={mockClose} location={{ x: 10, y: 20 }}/>
        </CharactersContext.Provider>
      </div>
    ));

    // there should be no option rendered
    expect(screen.getByRole('button', { name: 'juan' })).toBeInTheDocument();
  });

  test('Clicking on close button calls closeTarget function', async () => {
    const mockClose = jest.fn();
    render((
      <div>
        <CharactersContext.Provider value={['juan']}>
          <Choices closeTarget={mockClose} location={{ x: 10, y: 20 }} />
        </CharactersContext.Provider>
      </div>
    ));

    userEvent.click(screen.getByRole('button', { name: 'x' }));

    expect(mockClose).toBeCalled();
  });
})