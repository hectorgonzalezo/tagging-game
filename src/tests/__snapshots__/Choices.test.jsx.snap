// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Choices between characters Choices should be empty if context stays as initialized 1`] = `undefined`;
